### PR TITLE
chore: prework for supporting environments

### PR DESCRIPTION
**Pull request by mkramb**
* adding a top-level tree provider (which delegates to specific i.e. ```PipelineExplorer```)
* some small refactoring (moving all "commands" under common namespace)
* creating a base config file, next to ```extension.ts```

kapture 2018-05-30 at 10 53 40

### CODE REVIEW

File: src/explorer/pipelines/provider.ts [hunk 1]
```diff
@@ -28,7 +29,6 @@ export class PipelineExplorer {
             vscode.commands.registerCommand('PipelineExplorer.watchBuildLog', resource => this.watchBuildLog(resource)),
             vscode.commands.registerCommand('PipelineExplorer.startPipeline', resource => this.startPipeline(resource)),
             vscode.commands.registerCommand('PipelineExplorer.stopPipeline', resource => this.stopPipeline(resource)),
-            vscode.commands.registerCommand('PipelineExplorer.revealResource', () => this.reveal()),
```
Comment (mkramb) [inline src/explorer/pipelines/provider.ts]: Removed - was not used (should be referenced in ```package.json```)

**Comment by rawlingsj** [comment]
looks great @mkramb!

File: src/explorer/provider.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            return this.pipelineProvider.getTreeItem(element);
+        }
+
+        return element;
+    }
+
+	public getChildren(element?: ExplorerModel): ExplorerModel[] | Thenable<ExplorerModel[]> {
+        if (!element) {
+            return [
+                // new ExplorerModel('Environments', ExplorerContext.Environments),
```
Comment (mkramb) [inline src/explorer/provider.ts]: ```Environments``` comment-out until specific provider is implemented.

**Comment by mkramb** [comment]
@jstrachan if you can take a look if it aligns with your vision for environments -  https://github.com/jenkins-x/vscode-jx-tools/issues/30

**Comment by jstrachan** [comment]
So the Previews folder looks more like the Pipelines folder in that animation right? 

I was thinking we'd have a tree like this...

```
Jenkins X
  Environments
    Staging
    Production
    Previews
      myorg/
       myapp/
         PR-1
         PR-2
      another/
        PR-4
  Pipelines
    myorg/
      myapp/
        master/
        Pull Requestst
```

etc so you can either look at environments / preview environments or look at pipelines

**Comment by jstrachan** [comment]
those refactorings LGTM though!

**Comment by mkramb** [comment]
yes, ```enviroments``` would be a separate folder and based on https://github.com/jenkins-x/vscode-jx-tools/issues/30
existing content (pipelines) is under ```previews``` - is my thinking correct?

This is how it looks if I uncomment ```Enviroments``` - just a single node for now:
kapture 2018-05-30 at 11 28 34

**Comment by mkramb** [comment]
Renamed ```Previews``` to ```Pipelines```

**Comment by jstrachan** [comment]
yeah `Pipelines` is maybe a better name than `Previews`. 

I guess when it comes to `Preview Environments` we could keep them inside the Pipelines tree as they are right now and just make it easy to open the Pull Request / Preview App from the tree - it might be confusing having Previews show up inside `Environments` and `Pipelines` maybe?

**Comment by jstrachan** [comment]
right now the Environment resource for a Preview Environment has details of the Pull Request and application URL - those are not always available on the Pull Request node in the Pipelines/org/repo/PullRequests folder

**Comment by mkramb** [comment]
Ok, all good with merging this?
I know is not the final vision.

Next iteration is to populate environments. We can initially go with having ```preview environments``` also inside ```environments``` (I agree it could be confusing) and we can later decide how to best organize the structure.

**Pull request merged**